Formula/bpytop.rb

```diff
@@ -18,6 +18,9 @@ class Bpytop < Formula
   on_macos do
     depends_on "osx-cpu-temp"
   end
+  on_linux do
+    depends_on "rust" => :build # for cryptography
```

Comment:
But it doesn't have a `cryptography` resource?

Reply:
This is a tricky one to figure out.  It looks like the upstream repo uses `poetry` to build it, and there seem to be a bunch of dependencies needed only at build time (possible some for building `poetry` itself?) which are not installed into the cellar.  I'm not that familiar with Python build systems, especially not `poetry` so any insights into how to figure out which dependencies we actually need and how to extract them would be much appreciated!

Reply:
You can try `brew update-python-resources bpytop`

Reply:
Hmm, I tried that and it still claims this only depends on `psutil`.  I also get the same result when using `homebrew-pypi-poet` after installing into a virtualenv.  I think `psutil` is the only *run-time* dependency for `bpytop`, but it has a bunch of *build-time* dependencies that are declared separately and installed into `/tmp`.  That might explain how it evades detection by both `brew update-python-resources` and `homebrew-pypi-poet`.  

I'm still having difficulty figuring what the build-time dependencies are from the upstream repo: https://github.com/aristocratos/bpytop.  In principle if we can figure out what they are, we could manually add the resources and they should be installed into the virtualenv.  Even if we still get a build time dependency on Rust, at least it would be declared explicitly.

Reply:
Should definitely be related to setting up `poetry` (build-time).
`cryptography` in lock file: https://github.com/python-poetry/poetry/blob/1.1.7/poetry.lock#L182-L188

Maybe we should try avoiding `virtualenv_install_with_resources` here (which is able to pull in arbitrary packages online).

Perhaps try creating the virtualenv, install `psutil` into it, then run the Makefile?
https://github.com/aristocratos/bpytop#manual-installation-linux-osx-and-freebsd
```
git clone https://github.com/aristocratos/bpytop.git
cd bpytop
sudo make install
```

May need to check shims.

Reply:
Does using `make` still download anything? Or are the other Python dependencies bundled?

Reply:
The only Python dependency is `psutil`. The Makefile is just a set of `mkdir`, `cp`, etc. commands.

Reply:
Where does `cryptography` come in then?

Reply:
I think `pip` will look at `bpytop`'s [pyproject.toml](https://github.com/aristocratos/bpytop/blob/master/pyproject.toml) according to [PEP 517](https://www.python.org/dev/peps/pep-0517/)
 and see `build-system` set to `poetry >= 0.12`, so it will try to install `poetry`.

One example dependency chain to install `poetry`:
`poetry -> keyring -> secretstorage -> cryptography`
Related: https://github.com/python-poetry/poetry/issues/3661#issuecomment-777257358

Reply:
I should have made some gist-logs of this but @cho-m is right.  The `cryptography` dependency comes from trying to install `poetry`, not from installing `bpytop` itself.  I couldn't figure out how to bypass `poetry` but I think @cho-m is on the right track with using the Makefile instead.

Reply:
I got this to work with the Makefile, but now it can't find `psutil` unless I manually set `PYTHONPATH`.  I will push the fix as is so others can look at it.

Reply:
Most likely shim paths / virtualenv issue.
Options would be:
1. rewrite shim path to point to virtualenv python (e.g. `yelp-tools`)
2. use `write_env_script` to add `PYTHONPATH`. Could consider dropping virtualenv and just installing into directory (e.g. `pipx`)

Reply:
Thanks for those suggestions @cho-m.  I used the first one and it worked great!  I've pushed the changes now.